src/recipes/editor-plugin/inspector-plugins.md

```diff
@@ -0,0 +1,342 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Inspector plugins
+
+The inspector dock allows you to create custom widgets to edit properties through plugins.
+This can be beneficial when working with custom datatypes and resources, although you can
+use the feature to change the inspector widgets for built-in types. You can design custom
+controls for specific properties, entire objects, and even separate controls associated
+with particular datatypes.
+For more info, see
+[docs.godotengine.org](https://docs.godotengine.org/en/stable/classes/class_editorinspectorplugin.html#class-editorinspectorplugin).
+
+The [example](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/inspector_plugins.html) in the Godot docs in Rust.
+
+Before:
+
+![Before](./images/before.png)
+
+After:
+
+![After](./images/after.png)
+
+Add this dependency to Rust with the shell in the same directory as `Cargo.toml`.
+
+```bash
+cargo add rand
+```
+
+Add file `addon.rs` and import it in `lib.rs`:
+
+```rust
+// file: lib.rs
+mod addon;
+```
+
+Add the following imports at the beginning of the file
+
+```rust
+use godot::{
+    engine::{
+        Button, EditorInspectorPlugin, EditorPlugin, EditorProperty, IEditorInspectorPlugin,
+        IEditorPlugin, IEditorProperty,
+    },
+    global,
+    prelude::*,
+};
+use rand::Rng;
+```
+
+Since Rust is a statically typed language, we will proceed in reverse order unlike in Godot documentation, to avoid encountering errors unnecessarily.
+
+
+## Add Property Editor
+
+To begin with, let's define the editor for properties:
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorProperty)]
+struct RandomIntEditor {
+    base: Base<EditorProperty>,
+    button: Option<Gd<Button>>,
+}
+```
+
+After that, we need to add an implementation for the trait `IEditorProperty`:
+
+```rust
+#[godot_api]
+impl IEditorProperty for RandomIntEditor {
+    fn enter_tree(&mut self) {
+        // Create button element
+        let mut button = Button::new_alloc();
+        // Add handler for this button, handle_press will be define in another impl
+        button.connect(
+            "pressed".into(),
+            self.base().callable("handle_press".to_godot()),
+        );
+        button.set_text("Randomize".into());
+        // Save pointer to the button into struct
+        self.button = Some(button.clone());
+        self.base_mut().add_child(button.to_variant().to());
```

Comment:
Code style:
- Comments should end with full stops.
- Before comments, add empty line, so it's clear what belongs together.
```suggestion
        // Create button element.
        let mut button = Button::new_alloc();
        
        // Add handler for this button, handle_press will be define in another impl.
        button.connect(
            "pressed".into(),
            self.base().callable("handle_press".to_godot()),
        );
        button.set_text("Randomize".into());
        
        // Save pointer to the button into struct.
        self.button = Some(button.clone());
        self.base_mut().add_child(button.to_variant().to());
```